pkg/frr/frr.go

```diff
@@ -47,6 +63,76 @@ func (h *ModulefrrHandler) HandleEvent(eventType string, objectData *eventbus.Ob
 	}
 }
 
+// HandleAction handles the actions
+func (h *ModuleFrrActionHandler) HandleAction(actionType string, actionData *actionbus.ActionData) {
+	switch actionType {
+	case "preReplay":
+		log.Printf("Module FRR received %s\n", actionType)
+		handlePreReplay(actionData)
+	default:
+		log.Printf("error: Unknown action type %s", actionType)
+	}
+}
+
+func handlePreReplay(actionData *actionbus.ActionData) {
+	var deferErr error
+
+	defer func() {
+		// The ErrCh is used in order to notify the sender that the preReplay step has
+		// been executed successfully.
+		actionData.ErrCh <- deferErr
+	}()
+
+	// Backup the current running config
+	if err := os.Rename(runningFrrConfFile, backupFrrConfFile); err != nil {
+		log.Printf("FRR: handlePreReplay(): Failed to backup running config of FRR: %s\n", err)
+		deferErr = err
+		return
+	}
+
+	// Create a new running config based on the basic/initial FRR config
+	input, err := os.ReadFile(basicFrrConfFile)
+	if err != nil {
+		log.Printf("FRR: handlePreReplay(): Failed to read content of %s: %s\n", basicFrrConfFile, err)
+		deferErr = err
+		return
+	}
+
+	if err := os.WriteFile(runningFrrConfFile, input, 0600); err != nil {
+		log.Printf("FRR: handlePreReplay(): Failed to write content to %s: %s\n", runningFrrConfFile, err)
+		deferErr = err
+		return
+	}
+
+	// Change ownership of the frr.conf to frr:frr
+	group, err := user.Lookup("frr")
+	if err != nil {
+		log.Printf("FRR: handlePreReplay(): Failed to lookup user frr %s\n", err)
+		deferErr = err
+		return
+	}
+
+	uid, _ := strconv.Atoi(group.Uid)
+	gid, _ := strconv.Atoi(group.Gid)
+
+	if err := os.Chown(runningFrrConfFile, uid, gid); err != nil {
+		log.Printf("FRR: handlePreReplay(): Failed to chown of %s to frr:frr : %s\n", runningFrrConfFile, err)
+		deferErr = err
+		return
+	}
+
+	// Restart FRR daemon
+	_, errCmd := utils.Run([]string{"systemctl", "restart", "frr"}, false)
+	if errCmd != 0 {
+		log.Println("FRR: handlePreReplay(): Failed to restart FRR daemon")
+		err := fmt.Errorf("restart FRR daemon failed")
+		deferErr = err
+		return
+	}
+
+	log.Println("FRR: handlePreReplay(): The pre-replay procedure has executed successfully")
+}
```

Comment:
The whole reason that we do the replay procedure is because the frr configuration is malformed for whatever reason and we want to repalce it with a clean base configuration and replay the db obects in order to reapply the dynamic configuration. So there is no point to restore the previous configuration in case of error as this malformed configuration was the reason that the replay procedure started in first place.

Reply:
I mean if we are going to restore renamed configuration files and etc.

Reply:
for me it sounds strange

Reply:
I mean what is the point of restoring the renamed configuration files in case of error ? Those old configuration files that has been renamed are malformed (this is the reason that we are doing the replay procedure). So even if we restore them in case of some error we will still have malformed files so not sure what we gain with that.

Reply:
Let's assume that we restore the old configuration in case of error. How do you suggest to do it ?